pydoctor/model.py

```diff
@@ -846,6 +841,7 @@ def setSourceHref(self, mod: _ModuleT, source_path: Path) -> None:
             mod.sourceHref = None
         else:
             projBaseDir = mod.system.options.projectbasedirectory
+            assert projBaseDir is not None
```

Comment:
Yes, OK. If option self.options.htmlsourcebase if not None, options.projectbasedirectory is Path for sure, because Options() class fails other wise, so that's ok.